labs/lab6/lab6.adoc

```diff
@@ -29,56 +37,311 @@ Password: INSTRUCTOR WILL PROVIDE
 ----
 
-== Expand the Cluster
+The application itself is located in a file named link:../alertmanager-tower-bridge/bridge[bridge.py] which is Python based and creates a small HTTP based web server which accepts requests in the JSON format AlertManager is expected to send. After parsing the input, the application executes a POST request to Ansible Tower to execute the Job Template. Also included in the application is a set of custom link:https://docs.openshift.com/container-platform/latest/creating_images/s2i.html#s2i-scripts[Source to Image scripts] that dictate how the application should be built and run.
 
-The _0-Self-Configure_ job template executed earlier configured a set of templates to automate the process to expand the OpenShift cluster.
+After browsing through the application, change directories to the _alertmanager-tower-bridge-apb_ directory where we will walkthrough the process of creating an Ansible Playbook Bundle.
 
-Using a web browser from the student workstation machine, navigate to the Ansible Tower instance:
+The link:https://access.redhat.com/documentation/en-us/openshift_container_platform/3.9/html-single/ansible_playbook_bundle_development_guide/index[Ansible Playbook Development Guide] provides a detailed overview for developing Ansible Playbook Bundles. The majority of the functionality has already been provided for you.  Take a moment to navigate through the link:https://access.redhat.com/documentation/en-us/openshift_container_platform/3.9/html-single/ansible_playbook_bundle_development_guide/index#apb-devel-writing-ref-directory[folder structure] including the APB spec file (apb.yml) and the directories containing Ansible link:http://docs.ansible.com/ansible/latest/user_guide/playbooks.html[playbooks] and link:https://docs.ansible.com/ansible/devel/user_guide/playbooks_reuse_roles.html[roles].
 
-link:https://tower-<student_id>.labs.sysdeseng.com[https://tower-<student_id>.labs.sysdeseng.com] 
+Build the apb which will execute a Docker build using the _Dockerfile_ located within the _alertmanager-tower-bridge-apb_ directory.
+
+.master$
+[source, bash]
+----
+apb build
+----
+
+A new image containing the APB was created locally. This can be confirmed by listing all of the images on the machine and locating the image called _alertmanager-tower-bridge-apb_.
+
+.master$
+[source, bash]
+----
+docker images
+----
+
+The final step in the Ansible Playbook Bundle creation process is to push the image from the local machine to OpenShift's internal registry. In order to communicate with the registry, an authenticated user to the platform must be used as they contain an OAuth token needed to facilitate the communication.
+
+Login as the "<student_id>-admin" user which has elevated cluster privileges:
+
+.master$
+[source, bash]
+----
+oc login -u <student_id>-admin
+----
+
+Push the APB to the OpenShift integrated registry
+
+.master$
+[source, bash]
+----
+apb push
```

Comment:
need sudo here:

```$ apb push
Didn't find OpenShift Ansible Broker route in namespace: ansible-service-broker.                    Trying openshift-ansible-service-broker
version: 1.0
name: alertmanager-tower-bridge-apb
description: Bridge Application between AlertManager and Ansible Tower
bindable: False
async: optional
metadata:
  displayName: AlertManager Ansible Tower Bridge
  console.openshift.io/iconClass: icon-ansible
plans:

<SNIP by scollier>

      - name: NAMESPACE
        title: OCP namespace to deploy the AlertManger Bridge
        required: true
        default: openshift-metrics
        type: string

Found registry IP at: 172.30.150.147:5000
Finished writing dockerfile.
Building APB using tag: [172.30.150.147:5000/openshift/alertmanager-tower-bridge-apb]
Error accessing the docker API. Is the daemon running?
Exception occurred! Error while fetching server API version: ('Connection aborted.', error(13, 'Permission denied'))
```

Reply:
that doesn't work either, need to have them sudo first, then oc login.  then, make sure clone is done as root.